### PR TITLE
Stories

### PR DESCRIPTION
**Pull request by khaldoon9kh**
1- Single story card
2- Single Blog page
3- Fetch posts from WordPress API
4- still have problems with Router to Single Blog Page.

image

### CODE REVIEW

File: src/components/Blog/index.css
Comment (fadime-ozdemir) [inline src/components/Blog/index.css]: Please, put them as a className in the component. Ex: d-flex

File: src/components/Blog/index.css
Comment (fadime-ozdemir) [inline src/components/Blog/index.css]: If we don't need the code, it's better to delete them

File: src/components/Blog/index.jsx
Comment (fadime-ozdemir) [inline src/components/Blog/index.jsx]: We shouldn't leave console.logs when deploying

File: src/components/Blog/index.jsx
Comment (fadime-ozdemir) [inline src/components/Blog/index.jsx]: XD I like the name

File: src/components/SingleFeature/index.css
Comment (fadime-ozdemir) [inline src/components/SingleFeature/index.css]: Please, use variables for colors and rem instead of px

File: src/components/SingleFeature/index.css
Comment (fadime-ozdemir) [inline src/components/SingleFeature/index.css]: Please, use variables for colors and rem instead of px

File: src/components/SingleFeature/index.css
Comment (fadime-ozdemir) [inline src/components/SingleFeature/index.css]: Please, use variables for colors and rem instead of px

File: src/components/SingleStoryCard/index.css
Comment (fadime-ozdemir) [inline src/components/SingleStoryCard/index.css]: Please, use rem instead of em

File: src/components/SingleStoryCard/index.css
Comment (fadime-ozdemir) [inline src/components/SingleStoryCard/index.css]: Please, use variables for colors

File: src/components/SingleStoryCard/index.css
Comment (fadime-ozdemir) [inline src/components/SingleStoryCard/index.css]: Please, use variables for colors and rem instead of px

File: src/components/SingleStoryCard/index.css
Comment (fadime-ozdemir) [inline src/components/SingleStoryCard/index.css:35]: Please, use variables for colors and rem instead of px

**Review by fadime-ozdemir — CHANGES_REQUESTED** [review]
Overall I like how the blogs page looks! Good job!
There are some minor problems with the units and color codes.

File: src/components/Blog/index.jsx
Comment (halitbatur) [inline src/components/Blog/index.jsx]: I dont think this is done in the correct way, there should be something called useParams that u get the id from please check this page for further info https://reactrouter.com/web/example/url-params

File: src/components/Blog/index.jsx
Comment (halitbatur) [inline src/components/Blog/index.jsx]: this should have an empty [] inside of it in order to work only once the component gets renderd

**Review by halitbatur — CHANGES_REQUESTED** [review]

File: src/components/Blog/index.css
Comment (khaldoon9kh) [reply src/components/Blog/index.css]: fixed

File: src/components/Blog/index.css
Comment (khaldoon9kh) [reply src/components/Blog/index.css]: fixed

File: src/components/Blog/index.jsx
Comment (khaldoon9kh) [reply src/components/Blog/index.jsx]: removed

File: src/components/SingleFeature/index.css
Comment (khaldoon9kh) [reply src/components/SingleFeature/index.css]: edited

File: src/components/SingleFeature/index.css
Comment (khaldoon9kh) [reply src/components/SingleFeature/index.css]: revised

File: src/components/SingleStoryCard/index.css
Comment (khaldoon9kh) [reply src/components/SingleStoryCard/index.css]: changed

File: src/components/SingleStoryCard/index.css
Comment (khaldoon9kh) [reply src/components/SingleStoryCard/index.css]: changed

File: src/components/SingleStoryCard/index.css
Comment (khaldoon9kh) [reply src/components/SingleStoryCard/index.css]: changed

File: src/components/SingleStoryCard/index.css
Comment (khaldoon9kh) [reply src/components/SingleStoryCard/index.css:35]: changed

File: src/components/Blog/index.jsx
Comment (khaldoon9kh) [reply src/components/Blog/index.jsx]: So I tried using the same method, I succeeded in loading the page with the id. However, when I'm using "dangerouslySetInnerHTML", the page won't load as the fitch takes some time and it takes them as undefined.
how do you suggest fixing this issue? @halitbatur

File: src/components/Blog/index.jsx
Comment (khaldoon9kh) [reply src/components/Blog/index.jsx]: fixed

**Comment by khaldoon9kh** [comment]
I changed the styling accordingly.
However, I still have problems with loading Blog page, which causes Netlify errors.

Any suggestions on how to fix this issue?

Screen Shot 2020-11-10 at 5 45 55 PM

File: src/components/Blog/index.jsx
Comment (khaldoon9kh) [reply src/components/Blog/index.jsx]: every time I leave the [], i receive this error:
image

I will try to google it more to see how can I fix it.

Comment (halitbatur) [reply src/components/Blog/index.jsx]: @khaldoon9kh try to move the function definitions inside the useEffect

Comment (fadime-ozdemir) [reply src/components/Blog/index.jsx]: possible solutions: useCallback() or add the function also between brakets. More explanation in the below links.  
https://stackoverflow.com/questions/55840294/how-to-fix-missing-dependency-warning-when-using-useeffect-react-hook
https://stackoverflow.com/questions/58624200/react-hook-useeffect-has-a-missing-dependency-dispatch
https://dev.to/r3wt/useeffect-missing-dependency-need-advice-4d6b

**Review by fadime-ozdemir — CHANGES_REQUESTED** [review]
There are some problems with the tests.
If you know that you need some specific props I think it is better to destructor.

File: src/components/SingleStoryCard/index.jsx
Comment (fadime-ozdemir) [inline src/components/SingleStoryCard/index.jsx]: If you know that you need some specific props I think it is better to destructor.

File: src/components/SingleStoryCard/index.jsx
Comment (fadime-ozdemir) [inline src/components/SingleStoryCard/index.jsx]: If you don't need it, remember to delete!

File: src/containers/Stories/index.jsx
Comment (fadime-ozdemir) [inline src/containers/Stories/index.jsx]: If you don't need it, remember to delete!

File: src/containers/Stories/index.jsx
Comment (fadime-ozdemir) [inline src/containers/Stories/index.jsx]: If you don't need it, remember to delete!

**Review by fadime-ozdemir — CHANGES_REQUESTED** [review]

File: src/containers/Stories/index.jsx
Comment (khaldoon9kh) [inline src/containers/Stories/index.jsx]: This commented part is to filter the blogs with the translation. 
I commented it until we operate the i18n.

**Review by khaldoon9kh — COMMENTED** [review]
I fixed all errors.
Added LoadingPage.
Single Blog and Stories are fully working now.

Blog Header
image

Blog content
image

**Review by fadime-ozdemir — APPROVED** [review]
LGTM! Good Job!

File: src/components/SingleStoryCard/index.jsx
Comment (halitbatur) [reply src/components/SingleStoryCard/index.jsx]: @khaldoon9kh yes you c an destruct the props to make it easier for other developers to know what props are getting passed

File: src/containers/Blog/index.jsx
Comment (halitbatur) [inline src/containers/Blog/index.jsx]: no console.logs

File: src/containers/Blog/index.jsx
Comment (halitbatur) [inline src/containers/Blog/index.jsx]: lets move this const inside the useEffect since we only use it there and we want its scope as small as possible

File: src/containers/Stories/index.jsx
Comment (halitbatur) [inline src/containers/Stories/index.jsx]: no commented code

File: src/containers/Stories/index.jsx
Comment (halitbatur) [reply src/containers/Stories/index.jsx]: @khaldoon9kh we shouldn't merge any commented code if it's not used then u have to move to to another branch this branch should include only the work for stories

**Review by halitbatur — CHANGES_REQUESTED** [review]

File: src/containers/Blog/index.jsx
Comment (khaldoon9kh) [reply src/containers/Blog/index.jsx]: props deconstructed.
consols deleted.

File: src/containers/Blog/index.jsx
Comment (khaldoon9kh) [reply src/containers/Blog/index.jsx]: moved

File: src/containers/Stories/index.jsx
Comment (khaldoon9kh) [reply src/containers/Stories/index.jsx]: removed

File: src/containers/Blog/index.jsx
Comment (khaldoon9kh) [reply src/containers/Blog/index.jsx]: Moved

**Comment by khaldoon9kh** [comment]
@halitbatur I revised the code as you requested.

PTAL

File: src/components/SingleStoryCard/index.jsx
Comment (halitbatur) [inline src/components/SingleStoryCard/index.jsx]: now here's whats confusing me all the blogs are in a url which is /stories and the single story is in /blog/:id so please either change the stories components to be blogs or change the blog to be story, the same thing with the components name

**Review by halitbatur — CHANGES_REQUESTED** [review]

File: src/components/SingleStoryCard/index.jsx
Comment (halitbatur) [reply src/components/SingleStoryCard/index.jsx]: and merge it with the latest update of dev branch

**Comment by halitbatur** [comment]
PTAL

File: src/components/SingleStoryCard/index.jsx
Comment (khaldoon9kh) [reply src/components/SingleStoryCard/index.jsx]: Resolved

**Comment by khaldoon9kh** [comment]
@halitbatur 
Changed Blog component to Story.
PTAL

File: src/components/SingleStoryCard/index.css
Comment (fadime-ozdemir) [inline src/components/SingleStoryCard/index.css]: this should be a variable or just white instead of the # code so it is more readable

File: src/containers/Stories/index.jsx
Comment (fadime-ozdemir) [inline src/containers/Stories/index.jsx:11]: We should decide to use React.hook or directly the hook during the bug fix week

**Review by fadime-ozdemir — CHANGES_REQUESTED** [review]
Please, don't consider the comment on the react.hook for now, it is just a reminder for all of us

File: src/containers/Stories/index.jsx
Comment (khaldoon9kh) [reply src/containers/Stories/index.jsx:11]: I don't think it's that problomatic, but yes we should

File: src/components/SingleStoryCard/index.css
Comment (khaldoon9kh) [reply src/components/SingleStoryCard/index.css]: edited

**Comment by khaldoon9kh** [comment]
Errors fixed

PTAL

**Review by fadime-ozdemir — APPROVED** [review]
LGTM

**Review by halitbatur — APPROVED** [review]

**Pull request merged**
